eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -363,15 +367,37 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
    */
 
   when(NORMAL)(handleExceptions {
-    case Event(c: ForbiddenCommandDuringSplice, d: DATA_NORMAL) if d.spliceStatus != SpliceStatus.NoSplice =>
+    case Event(c: ForbiddenCommandDuringQuiescence, d: DATA_NORMAL) if d.spliceStatus.isInstanceOf[QuiescenceNegotiation] =>
+      val error = ForbiddenDuringQuiescence(d.channelId, c.getClass.getSimpleName)
+      c match {
+        case c: CMD_ADD_HTLC => handleAddHtlcCommandError(c, error, Some(d.channelUpdate))
+        case _: HtlcSettlementCommand => stay() // htlc settlement commands will be ignored and replayed when not quiescent
+        case _ => handleCommandError(error, c)
+      }
+
+    case Event(c: ForbiddenCommandDuringSplice, d: DATA_NORMAL) if d.spliceStatus != SpliceStatus.NoSplice  && !d.spliceStatus.isInstanceOf[QuiescenceNegotiation] =>
       val error = ForbiddenDuringSplice(d.channelId, c.getClass.getSimpleName)
       c match {
         case c: CMD_ADD_HTLC => handleAddHtlcCommandError(c, error, Some(d.channelUpdate))
-        // NB: the command cannot be an htlc settlement (fail/fulfill), because if we are splicing it means the channel is idle and has no htlcs
+        // NB: the command cannot be an htlc settlement (fail/fulfill), because if we are splicing without quiescence support it means the channel is idle and has no htlcs
         case _ => handleCommandError(error, c)
       }
 
-    case Event(msg: ForbiddenMessageDuringSplice, d: DATA_NORMAL) if d.spliceStatus != SpliceStatus.NoSplice && !d.spliceStatus.isInstanceOf[SpliceStatus.SpliceRequested] =>
+    case Event(msg: ForbiddenMessageDuringSplice, d: DATA_NORMAL) if d.commitments.params.useQuiescence && d.spliceStatus != SpliceStatus.NoSplice && !d.spliceStatus.isInstanceOf[SpliceStatus.InitiatorQuiescent] =>
```

Comment:
In change [1e7dfad](https://github.com/ACINQ/eclair/pull/2680/commits/1e7dfad4e6287a69843ea0efdb9e568b2f3a48be) I changed `NonInitiatorQuiescent` to extend `QuiescentSpliceStatus` instead of `QuiescenceNegotiation.NonInitiator`. Logically once the non-initiator sends `stfu` they are quiescent (no longer negotiating quiescence), even if the initiator might not have yet received their `stfu`. This change allowed me to simplify the tests for forbidden messages that I previously called "... after initiator receives stfu from remote" and rename them to "... while quiescent".

Because forbidden messages do not trigger a channel close, I could simplify the tests to ensure both alice and bob return a warning (and disconnect) in the same test and remove the separate tests for bob. 

NB: while renaming the tests, I found/removed a duplicate test called "recv settlement commands after initiator receives stfu from remote" which was identical to "recv forbidden commands while quiescent".